base/iobuffer.jl

```diff
@@ -291,9 +289,17 @@ end
             end
         end
     end
-    n = min(nshort + (io.append ? io.size : io.ptr-1), io.maxsize)
-    if n > length(io.data)
-        resize!(io.data, n)
+end
+
+@inline ensureroom(io::GenericIOBuffer, nshort::Int) = ensureroom(io, UInt(nshort))
+@inline function ensureroom(io::GenericIOBuffer, nshort::UInt)
+    if !io.writable || !io.seekable
```

Comment:
Can we make this `!io.writable || (!io.seekable && io.ptr > 1)`? One extra comparison and branch here shouldn't be much extra cost, but that would restore the expected fastpath for `!seekable`.